File: |-- src_R/|   |-- 06_interval_proporsi.R
# PROGRAM 6: INTERVAL KONFIDENSI PROPORSI
# FUNGSI
interval <- function(p, alpha, n) {
  # Validasi proporsi (sesuai soal: < 0 atau > 1)
  if (p < 0 || p > 1) {
    cat("STATUS: TIDAK VALID - Proporsi harus antara 0 dan 1\n")
    return()
  }
  # Validasi n harus positif
  if (n <= 0) {
    cat("STATUS: TIDAK VALID - Ukuran sampel harus positif\n")
    return()
  }
  # Tentukan nilai z berdasarkan alpha
  if (alpha == 0.05) {
    z <- 1.96
  } else if (alpha == 0.10) {
    z <- 1.645
  } else {
    cat("STATUS: TIDAK VALID - Alpha harus 0.05 atau 0.10\n")
    return()
  }
  # Hitung interval
  se <- sqrt((p * (1 - p)) / n)
  margin <- z * se
  bawah <- p - margin
  atas <- p + margin
  
  # Tampilkan hasil
  cat("Proporsi sampel (p^) =", p, "\n")
  cat("Alpha =", alpha, "\n")
  cat("n =", n, "\n")
  cat("Interval konfidensi:", round(bawah, 4), "< p <", round(atas, 4), "\n")
  cat("STATUS: VALID\n")
}
# SKENARIO 1 
interval(0.5, 0.05, 100)

# SKENARIO 2 
interval(0.7, 0.10, 50)

# SKENARIO 3 
interval(1.5, 0.05, 100)

File: |-- src_python/|   |-- 06_interval_proporsi.py
#PROGRAM 6: INTERVAL KONFIDENSI PROPORSI

import math

def interval(p, alpha, n):
    #Validasi proporsi (harus antara 0 dan 1)
    if p < 0 or p > 1:
        print("STATUS: TIDAK VALID - Proporsi harus antara 0 dan 1")
        return()
    
    #Validasi n harus positif
    if n <= 0:
        print("STATUS: TIDAK VALID - Ukuran sampel harus positif")
        return()
    
    #Tentukan nilai z berdasarkan alpha
    if alpha == 0.05:
        z = 1.96
    elif alpha == 0.10:
        z = 1.645
    else:
        print("STATUS: TIDAK VALID - Alpha harus 0.05 atau 0.10")
        return()
    
    #Hitung interval
    se = math.sqrt((p * (1 - p)) / n)
    margin = z * se
    bawah = p - margin
    atas = p + margin
    
    #Tampilkan hasil
    print(f"Proporsi sampel (p^) = {p}")
    print(f"Alpha = {alpha}")
    print(f"n = {n}")
    print(f"Interval konfidensi: {bawah:.4f} < p < {atas:.4f}")
    print("STATUS: VALID")


#SKENARIO 1 
print("SKENARIO 1")
interval(0.5, 0.05, 100)

#SKENARIO 2 
print("SKENARIO 2")
interval(0.7, 0.10, 50)

#SKENARIO 3 
print("SKENARIO 3")
interval(1.5, 0.05, 100)

#SKENARIO 4
print("SKENARIO 4")
interval(1, 0.10, 50)
